src/main/java/uk/gov/hmcts/reform/sscs/callback/handlers/IssueGenericLetterHandler.java

```diff
@@ -0,0 +1,291 @@
+    }
+
+    @Override
+    public DispatchPriority getPriority() {
+        return DispatchPriority.LATEST;
+    }
+
+    @Override
+    public void handle(CallbackType callbackType, Callback<SscsCaseData> callback) {
+        if (!canHandle(callbackType, callback)) {
+            log.info("Cannot handle this event for case id: {}", callback.getCaseDetails().getId());
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        SscsCaseData caseData = callback.getCaseDetails().getCaseData();
+        long id = callback.getCaseDetails().getId();
+
+        docmosisTemplate = getDocmosisTemplate(caseData);
+        docmosisCoverSheetTemplate = getDocmosisCoverSheet(caseData);
+
+        process(id, caseData);
+    }
+
+    private void sendLetter(long caseId, SscsCaseData caseData, List<Pdf> pdfs) {
+        log.info("Sending letter");
+        Optional<UUID> id = bulkPrintService.sendToBulkPrint(pdfs, caseData);
+        //Optional.of(UUID.randomUUID());
+
+        Pdf letter = pdfs.get(0);
+
+        if (id.isPresent()) {
+            ccdNotificationService.storeNotificationLetterIntoCcd(EventType.ISSUE_GENERIC_LETTER, letter.getContent(),
+                caseId);
+            log.info("Generic letters were send for case {}, send-letter-service id {}", caseId, id.get());
+        } else {
+            log.error("Failed to send to bulk print for case {}. No print id returned", caseId);
+        }
+    }
+
+    private String getDocmosisTemplate(SscsCaseData caseData) {
+        return docmosisTemplateConfig.getTemplate().get(caseData.getLanguagePreference())
+            .get("generic-letter").get("name");
+    }
+
+    private String getDocmosisCoverSheet(SscsCaseData caseData) {
+        return docmosisTemplateConfig.getTemplate().get(caseData.getLanguagePreference())
+            .get("generic-letter").get("cover");
+    }
+
+    private void process(long caseId, SscsCaseData caseData) {
+        log.info("Process the issue generic letter for the case : " + caseId);
+        List<Pdf> documents = new ArrayList<>();
+
+        if (YesNo.isYes(caseData.getAddDocuments())) {
+            documents.addAll(coverLetterService.getSelectedDocuments(caseData));
+        }
+
+        if (YesNo.isYes(caseData.getSendToAllParties())) {
+            sendToAllParties(caseId, caseData, documents);
+            return;
+        }
+
+        if (YesNo.isYes(caseData.getSendToApellant())) {
+            sendToAppellant(caseId, caseData, documents);
+        }
+
+        if (YesNo.isYes(caseData.getSendToRepresentative())) {
+            sendToRepresentative(caseId, caseData, documents);
+        }
+
+        if (YesNo.isYes(caseData.getSendToJointParty())) {
+            sendToJointParty(caseId, caseData, documents);
+        }
+
+        if (YesNo.isYes(caseData.getSendToOtherParties())) {
+            sendToOtherParties(caseId, caseData, documents);
+        }
+
+        // blank page??
```

Comment:
if its something to be done leave TODO comment rather then what you have got now